src/include/distributed/commands.h

```diff
@@ -477,6 +482,8 @@ extern void ErrorIfAnyPartitionRelationInvolvedInNonInheritedFKey(List *relation
 extern void DropRelationForeignKeys(Oid relationId, int flags);
 extern void ExecuteAndLogDDLCommandList(List *ddlCommandList);
 extern void ExecuteAndLogDDLCommand(const char *commandString);
+extern void ExecuteForeignKeyCreateCommandList(List *ddlCommandList,
```

Comment:
could be better if can make use of this function in `CreateCitusLocalTable` when executing shell table ddl events, but it wouldn't hurt for now as we only allow self referencing foreign keys when cascade_via_fkeys=false

Reply:
Agreed, we should tackle that separately. Do we have an issue for Citus local table limitations? Maybe add this item ?

Reply:
>Do we have an issue for Citus local table limitations? 

yes, I can this to there